src/nvc.rs

```diff
@@ -2,6 +2,100 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone)]
+pub struct NvcCacheEntry {
```

Comment:
nit: I think this should be called `CacheEntry` without the `Nvc` prefix. This would make the fully qualified name `nvc::CacheEntry` rather than `nvc::NvcCacheEntry`.